test/lib/pmd/PmdCatalogWrapper.test.ts

```diff
@@ -45,19 +45,20 @@ describe('PmdCatalogWrapper', () => {
 					const target = await TestablePmdCatalogWrapper.create({});
 					const params = (await target.buildCommandArray())[1];
 
-					// Expect there to be 6 parameters.
-					expect(params.length).to.equal(6, `Should have been 6 parameters: ${params}`);
-					// Expect the first two parameters to be the catalog and -cp flag.
-					expect(params[0]).to.equal('-DcatalogName=PmdCatalog.json', 'First parameter should be catalog override');
-					expect(params[1]).to.equal('-cp', 'Second parameter should be -cp flag');
+					// Expect there to be 7 parameters.
+					expect(params.length).to.equal(7, `Should have been 7 parameters: ${params}`);
+					// Expect the first three parameters to be the catalogHome, catalogName and -cp flag.
+					expect(params[0]).to.contain('-DcatalogHome=', 'First parameter should be catalog home path');
+					expect(params[1]).to.equal('-DcatalogName=PmdCatalog.json', 'Second parameter should be catalog override');
+					expect(params[2]).to.equal('-cp', 'Second parameter should be -cp flag');
```

Comment:
The unit testing cookbook suggests that assert messages and comments cause a maintenance burden that often isn't worth the effort. This test seems to show some of the issues they were afraid of. The comments and assert messages are out of sync with the code. If we leave them, maybe they can be rewritten not to include `Second`, `Third` etc. The reader can rely on the index passed into the params array. https://confluence.internal.salesforce.com/display/TESTSFDC/Unit+Testing+Best+Practices#UnitTestingBestPractices-Bejudiciouswithcustomassertionmessages

Reply:
Great point. Let me update the tests.